main/src/library/Map.flix

```diff
@@ -126,83 +87,72 @@ namespace Map {
     /// Otherwise, updates `m` with `k -> v`.
     ///
     pub def insertWithKey(f: (k, v, v) -> v & e, k: k, v: v, m: Map[k, v]): Map[k, v] & e =
-        if (memberOf(k, m)) adjustWithKey((a, b) -> f(a, v, b), k, m)
-        else
-            let Map(xs) = m;
-                Map((k, v) :: xs)
-
-
-
-
-
+        let Map(xs) = m;
+        match get(k, m) {
```

Comment:
This seems needlessly wasteful.

Reply:
Yeah, the performance of this did cross my mind while I was implementing it.
I'm considering modifying RBTree.insert to take an update function, e.g. `f: (k, v, v) -> v`.
Does that sound reasonable to you, or did you have something simpler in mind?

Reply:
I've refactored RBTree.insertHelper as RBTree.insertWith (used by the insert and insertWith methods) and added an RBTree.updateWith method (used by the adjust and update methods) to solve this.